components/global/user_guide_dropdown/user_guide_dropdown.tsx

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {Tooltip} from 'react-bootstrap';
+import {FormattedMessage, injectIntl, IntlShape} from 'react-intl';
+import IconButton from '@mattermost/compass-components/components/icon-button';
+
+import MenuWrapper from 'components/widgets/menu/menu_wrapper';
+import Menu from 'components/widgets/menu/menu';
+import OverlayTrigger from 'components/overlay_trigger';
+import {toggleShortcutsModal} from 'actions/global_actions';
+import {trackEvent} from 'actions/telemetry_actions';
+
+const askTheCommunityUrl = 'https://mattermost.com/pl/default-ask-mattermost-community/';
+
+type Props = {
+    intl: IntlShape;
+    helpLink: string;
+    reportAProblemLink: string;
+    enableAskCommunityLink: string;
+};
+
+type State = {
+    buttonActive: boolean;
+};
+
+class UserGuideDropdown extends React.PureComponent<Props, State> {
+    constructor(props: Props) {
+        super(props);
+        this.state = {
+            buttonActive: false,
+        };
+    }
+
+    toggleShortcutsModal = (e: MouseEvent) => {
+        e.preventDefault();
+        toggleShortcutsModal();
+    }
+
+    buttonToggleState = (menuActive: boolean) => {
+        this.setState({
+            buttonActive: menuActive,
+        });
+    }
+
+    askTheCommunityClick = () => {
+        trackEvent('ui', 'help_ask_the_community');
+    }
+
+    renderDropdownItems = (): React.ReactNode => {
```

Comment:
Whenever I see `render` in a function name for a component I always ask myself if this should just be a separate component or just in-lined.

Reply:
true, this will be replaced with the menu common component soon as well so we didn't focus on optimisation, just moving functionality.